roles/pulp_database/tasks/install_postgres.yml

```diff
@@ -1,29 +1,111 @@
 ---
 - block:
+    - name: Check if PostgreSQL < 10 is initialized.
+      stat:
+        path: "{{ postgresql_previous_data_dir }}/PG_VERSION"
+      register: previous_pgdata_dir_version
+
+    - name: Get PG_VERSION
+      command: "cat {{ postgresql_previous_data_dir }}/PG_VERSION"
+      register: pg_version
+      when: previous_pgdata_dir_version.stat.exists
+      tags:
+        - molecule-idempotence-notest
+
+    - name: Stop PostgreSQL < 10 service
+      systemd:
+        name: "{{ postgresql_previous_daemon }}"
+        state: stopped
+        enabled: no
+      register: pgservice_result
+      failed_when: "pgservice_result is failed and 'Service is in unknown state' not in pgservice_result.msg"
```

Comment:
Why do we need this failed_when condition?

Also, are you sure that consulting the registered variable works? I thought it always returns success. I tried to do this to add additional failed_when conditions on top of the module's built in, and it didn't work.

Reply:
I forgot to remove it